### DISCUSSION TITLE
SEI disappear when active material loss

### DISCUSSION THREAD

**Discussion by Yuxiao-Lan** [pybamm-team/PyBaMM#4428] [Q&A] [answered]
Hello pybamm!
I am running [coupled-degradation example](https://docs.pybamm.org/en/stable/source/examples/notebooks/models/coupled-degradation.html) and modified the following parameters:

```python
param["Outer SEI solvent diffusivity [m2.s-1]"] = 2.5000000000000002e-22 * 20
param["Negative electrode LAM constant proportional term [s-1]"] = 1.2e-6 * 20
```

After running 365 cycles, I found that the trend of porosity versus throughput capacity declines initially and then increases, as shown in the Figure. I am curious about which factors causes the effect.
In the `reaction_driven_porosity.py` file, line 60, the code illustrates how PyBaMM calculates the solid volume fraction, $\epsilon_s$:

$$
\epsilon_s = \epsilon_{init} - a_kL_{tot}
$$

According to this, porosity will increase as $a_k$ decreases due to the loss of active material. This is somewhat counterintuitive to me because the **SEI mass generated before is disappearing**. How can I modify the equation to represent the change in porosity over time as follows?

$$
\frac{d\epsilon_s}{dt} = - a_kdL_{tot}
$$

I would greatly appreciate any insights or assistance in clarifying this matter!

![image](https://github.com/user-attachments/assets/1b3e89d8-d02e-4b69-8523-481dcc1fa5ac)

**Comment by DrSOKane** [comment, marked as answer]
Hi @Yuxiao-Lan, I am aware of this issue and have proposed a fix. However, the fix caused an unexpected problem for models such as SPM where the option`"x-average side reactions"` is set to `"true"`. We have been trying to track down the source of the bug for months now, with no success.

**Reply by Yuxiao-Lan** [reply]
Thank you for the rapid response and explanation.
